src/serverenvironment.cpp

```diff
@@ -818,6 +836,14 @@ class ABMHandler
 		{
 			const MapNode &n = block->getNodeUnsafe(p0);
 			content_t c = n.getContent();
+			if (!block->contents_cached && !block->do_not_cache_contents) {
+				block->contents.insert(c);
+				if (block->contents.size() > 32) {
```

Comment:
does it make sense for this to be configurable?

Reply:
IMHO. We should find a good default. 64 turns out to be almost universally good.